packages/protobuf/src/reflect/reflect.ts

```diff
@@ -189,12 +189,12 @@ class ReflectMessageImpl<Desc extends DescMessage> implements ReflectMessage {
   set<Field extends DescField>(
     field: Field,
     value: ReflectSetValue<Field>,
-  ): FieldError | undefined {
+  ): void {
     assertOwn(this.message, field);
     if (this.check) {
       const err = checkField(field, value);
       if (err) {
-        return err;
+        throw err;
```

Comment:
We could throw in `checkField`, but the function is internal, so we can punt on it until there is a really good reason do change the function.